packages/ckeditor5-media-embed/src/mediaembedui.ts

```diff
@@ -40,74 +39,95 @@ export default class MediaEmbedUI extends Plugin {
 		const editor = this.editor;
-		const t = editor.t;
-		const button = dropdown.buttonView;
-		const registry = editor.plugins.get( MediaEmbedEditing ).registry;
-
-		dropdown.once( 'change:isOpen', () => {
-			const form = new ( CssTransitionDisablerMixin( MediaFormView ) )( getFormValidators( editor.t, registry ), editor.locale );
+		const locale = editor.locale;
+		const buttonView = new ButtonClass( editor.locale ) as InstanceType<T>;
+		const command = editor.commands.get( 'mediaEmbed' )!;
+		const t = locale.t;
+		const dialogPlugin = this.editor.plugins.get( 'Dialog' );
 
-			dropdown.panelView.children.add( form );
+		buttonView.set( {
+			label: t( 'Insert media' ),
+			icon: mediaIcon,
+			isToggleable: true
+		} );
 
-			// Note: Use the low priority to make sure the following listener starts working after the
-			// default action of the drop-down is executed (i.e. the panel showed up). Otherwise, the
-			// invisible form/input cannot be focused/selected.
-			button.on( 'open', () => {
-				form.disableCssTransitions();
+		buttonView.bind( 'isEnabled' ).to( command, 'isEnabled' );
 
-				// Make sure that each time the panel shows up, the URL field remains in sync with the value of
-				// the command. If the user typed in the input, then canceled (`urlInputView#fieldView#value` stays
-				// unaltered) and re-opened it without changing the value of the media command (e.g. because they
-				// didn't change the selection), they would see the old value instead of the actual value of the
-				// command.
-				form.url = command.value || '';
-				form.urlInputView.fieldView.select();
-				form.enableCssTransitions();
-			}, { priority: 'low' } );
-
-			dropdown.on( 'submit', () => {
-				if ( form.isValid() ) {
-					editor.execute( 'mediaEmbed', form.url );
-					editor.editing.view.focus();
-				}
-			} );
+		buttonView.on( 'execute', () => {
+			if ( dialogPlugin.id === 'mediaEmbed' ) {
+				dialogPlugin.hide();
 
-			dropdown.on( 'change:isOpen', () => form.resetFormStatus() );
-			dropdown.on( 'cancel', () => {
-				editor.editing.view.focus();
-			} );
+				return;
+			}
 
-			form.delegate( 'submit', 'cancel' ).to( dropdown );
-			form.urlInputView.fieldView.bind( 'value' ).to( command, 'value' );
+			this._showDialog();
+		} );
 
-			// Update balloon position when form error changes.
-			form.urlInputView.on( 'change:errorText', () => {
-				editor.ui.update();
-			} );
+		return buttonView;
+	}
 
-			// Form elements should be read-only when corresponding commands are disabled.
-			form.urlInputView.bind( 'isEnabled' ).to( command, 'isEnabled' );
-		} );
+	private _showDialog() {
+		const editor = this.editor;
+		const dialog = editor.plugins.get( 'Dialog' );
+		const command = editor.commands.get( 'mediaEmbed' )!;
+		const t = editor.locale.t;
 
-		dropdown.bind( 'isEnabled' ).to( command );
+		// if ( !this.formView ) {
```

Comment:
Why did you comment it out? I believe we do not want to create a new view each time.